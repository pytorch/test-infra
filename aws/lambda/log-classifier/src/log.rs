use once_cell::sync::Lazy;
use regex::Regex;
use std::collections::BTreeMap;

/// Representation of a single CI log for matching against.
#[derive(Debug)]
pub struct Log {
    /// Map of line number => line text. We use BTreeMap because we want:
    ///   - Reasonably efficient lookup by line number.
    ///   - Ordered traversal so that we can compute context ranges easily.
    ///   - Non-contiguous line numbers (e.g. if we skip some lines).
    pub lines: BTreeMap<usize, String>,
}

/// Matches ANSI escape sequences so that they can be stripped out of the log.
/// See: https://stackoverflow.com/questions/14693701/how-can-i-remove-the-ansi-escape-sequences-from-a-string-in-python
static ESCAPE_CODE_REGEX: Lazy<Regex> = Lazy::new(|| {
    Regex::new(r"(?:\x1B[@-Z\\-_]|[\x80-\x9A\x9C-\x9F]|(?:\x1B\[|\x9B)[0-?]*[ -/]*[@-~])").unwrap()
});

/// Matches the ISO8601 timestamp that GitHub Actions preprends to each log line
static TIMESTAMP_REGEX: Lazy<Regex> =
    Lazy::new(|| Regex::new(r"^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{7}Z ").unwrap());

impl Log {
    /// Create a log from a string, applying some preprocessing to make it
    /// easier to match against.
    pub fn new(log: String) -> Log {
        let mut lines = BTreeMap::new();
        let mut ignore_state = IgnoreStateMachine::new();

        // Do some preprocessing on the log lines.
        for (idx, raw_line) in log.lines().enumerate() {
            // GHA adds a timestamp to the front of every log. Strip it before matching.
            let line = TIMESTAMP_REGEX.replace(raw_line, "");

            // Strip ANSI escape codes that interfere with matching.
            let line = ESCAPE_CODE_REGEX.replace_all(&line, "");

            // If this line should be ignored, don't add it to the Log.
            if ignore_state.should_ignore(&line) {
                continue;
            }

            // Lines are 1-indexed!
            let line_number = idx + 1;
            lines.insert(line_number, line.into_owned());
        }

        Log { lines }
    }
}

/// Helper to manage the state for whether or not the matcher should be ignoring
/// the current line. We ignore matches against some regions of the logs are
/// known to be noisy or misleading.
#[derive(Debug)]
struct IgnoreStateMachine {
    is_ignoring: bool,
    start_ignore: Regex,
    stop_ignore: Regex,
}

impl IgnoreStateMachine {
    fn new() -> Self {
        let start_ignore =
            Regex::new(r"=================== sccache compilation log ===================").unwrap();
        let stop_ignore = Regex::new(r"=========== If your build fails, please take a look at the log above for possible reasons ===========").unwrap();
        Self {
            is_ignoring: false,
            start_ignore,
            stop_ignore,
        }
    }

    /// Check whether we should ignore the provided line, and advance the state
    /// machine one step.
    fn should_ignore(&mut self, line: &str) -> bool {
        if self.is_ignoring {
            if self.stop_ignore.is_match(line) {
                self.is_ignoring = false;
            }
            true
        } else {
            if self.start_ignore.is_match(line) {
                self.is_ignoring = true;
                true
            } else {
                false
            }
        }
    }
}
