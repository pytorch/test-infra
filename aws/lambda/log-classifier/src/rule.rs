use regex::Regex;
use serde::Deserialize;

/// Represents a single log classification rule for matching against.
#[derive(Debug, Clone)]
pub struct Rule {
    pub name: String,
    pub pattern: Regex,
    // If multiple rules match a log, the higher priority one wins. This value
    // is not manually set, it depends on the insertion order into RuleSet
    pub priority: u64,
}

/// Holds a set of Rules and manages their relative priority. Rules inserted
/// earlier have higher priority.
pub struct RuleSet {
    pub rules: Vec<Rule>,
}

/// Corresponds to the on-disk representation of `Rule` in ../ruleset.toml
#[derive(Debug, Clone, Deserialize)]
struct SerializedRule {
    name: String,
    pattern: String,
}

/// Corresponds to the on-disk representation of `RuleSet` in ../ruleset.toml
#[derive(Debug, Clone, Deserialize)]
struct SerializedRuleSet {
    #[serde(rename = "rule")]
    rules: Vec<SerializedRule>,
}

impl RuleSet {
    pub fn new() -> RuleSet {
        RuleSet { rules: Vec::new() }
    }

    pub fn new_from_config() -> RuleSet {
        let f = include_bytes!("../ruleset.toml");
        let config_rules: SerializedRuleSet = toml::from_slice(f).unwrap();

        let mut ret = RuleSet::new();
        for rule in config_rules.rules {
            ret.add(&rule.name, &rule.pattern);
        }

        ret
    }

    /// Add a rule to the ruleset.
    pub fn add(&mut self, name: &str, pattern: &str) {
        let lowest_priority = self.rules.last().map(|r| r.priority).unwrap_or(u64::MAX);
        let priority = lowest_priority.saturating_sub(1);

        self.rules.push(Rule {
            name: name.into(),
            pattern: pattern.parse().unwrap(),
            priority,
        })
    }
}
