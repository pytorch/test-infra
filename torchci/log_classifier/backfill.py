import argparse
import logging
import os
import sys
from urllib.request import urlopen

logger = logging.getLogger()
logger.setLevel(logging.INFO)

LAMBDA_URL = os.environ.get("LAMBDA_URL")
if LAMBDA_URL is None:
    raise RuntimeError("LAMBDA_URL not set, see --help")

ROCKSET_API_KEY = os.environ.get("ROCKSET_API_KEY")
if ROCKSET_API_KEY is None:
    raise RuntimeError("ROCKSET_API_KEY not set, see --help")


def send_lambda_request(id):
    logger.info(f"sending lambda request for id {id}")
    url = LAMBDA_URL + f"?{id}"
    with urlopen(url) as res:
        logger.info(f"lambda response for id {id}: {res.status}")


def do_backfill(n):
    # Import here to avoid requiring these dependencies in lambda
    from rockset import Client, Q, F, ParamDict
    from concurrent.futures import ThreadPoolExecutor, wait

    # query rockset for failed GHA job ids
    client = Client(
        api_key=ROCKSET_API_KEY,
        api_server="https://api.rs2.usw2.rockset.com",
    )
    qlambda = client.QueryLambda.retrieve(
        "unclassified", version="672227495ac70f7d", workspace="commons"
    )

    params = ParamDict()
    results = qlambda.execute(parameters=params).results
    # q = (
    #     Q("GitHub-Actions.workflow_job")
    #     .where(F["conclusion"] == "failure")
    #     .highest(n, F["_event_time"])
    #     .select(F["id"])
    # )
    # results = client.sql(q)
    ids = [result["id"] for result in results]
    with ThreadPoolExecutor() as executor:
        futures = []
        for id in ids:
            futures.append(executor.submit(send_lambda_request, id))
        wait(futures)

    logger.info("done!")


if __name__ == "__main__":
    logging.basicConfig(
        format="<%(levelname)s> [%(asctime)s] %(message)s",
        level=logging.INFO,
        stream=sys.stderr,
    )

    parser = argparse.ArgumentParser(
        description="""\
        Backfill classifications for failing jobs. This is useful when we add a
        new rule and want old jobs to use it.

        You need to set ROCKSET_API_KEY to a valid Rockset API key and
        LAMBDA_URL to the API Gateway URL for the ossci-log-analyzer lambda.

        This uses the lambda to backfill, NOT your local repo.
        """
    )
    parser.add_argument(
        "--n",
        type=int,
        default=1000,
        help="Classify the `n` most recent failed jobs",
    )
    args = parser.parse_args()

    do_backfill(args.n)
