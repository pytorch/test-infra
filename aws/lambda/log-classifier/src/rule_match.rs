use crate::log::Log;
use crate::rule::Rule;
use serde::Serialize;

/// Represents a successful match of a log line against a rule.
#[derive(Debug)]
pub struct Match {
    pub line_number: usize,
    pub rule: Rule,
    /// The capture groups in the regex pattern. If no capture groups were
    /// specified, this is the entire match.
    pub captures: Vec<String>,
}

/// The actual format that we insert to Rockset.
#[derive(Debug, Serialize)]
pub struct SerializedMatch {
    rule: String,
    line: String,
    line_num: usize,
    captures: Vec<String>,
}

impl SerializedMatch {
    pub fn new(m: &Match, log: &Log) -> SerializedMatch {
        // Unwrap because we know this is a valid key (since the Log object is never mutated.)
        let line = log.lines.get(&m.line_number).unwrap();
        SerializedMatch {
            rule: m.rule.name.clone(),
            line: line.clone(),
            line_num: m.line_number,
            captures: m.captures.clone(),
        }
    }
}
