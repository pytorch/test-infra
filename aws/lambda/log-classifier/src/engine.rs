use crate::log::Log;
use crate::rule::{Rule, RuleSet};
use crate::rule_match::Match;
use rayon::prelude::*;

/// Evaluate `rule` against `log`. Returns a `Match` if there was a successful
/// match, otherwise None.
pub fn evaluate_rule(rule: &Rule, log: &Log) -> Option<Match> {
    // Iterate in reverse order, as later log lines are more likely to be
    // interesting to us.
    for (line_number, line) in log.lines.iter().rev() {
        if let Some(captures) = rule.pattern.captures(line) {
            let captures = if captures.len() == 1 {
                // If there is only one capture, it means that the regex itself
                // had no capturing groups (since the first capture group is
                // always the whole match). In that case, just record the whole
                // match as a capture.
                vec![captures.get(0).unwrap().as_str().to_string()]
            } else {
                captures
                    .iter()
                    .skip(1) // the first capture is the whole match, so skip it
                    .flatten() // remove non-matching captures
                    .map(|c| c.as_str().to_string())
                    .collect()
            };

            return Some(Match {
                line_number: *line_number,
                rule: rule.clone(),
                captures,
            });
        }
    }
    None
}

/// Evaluate the ruleset against `log`. Returns the highest-priority match, or
/// None if no rule matched.
pub fn evaluate_ruleset(ruleset: &RuleSet, log: &Log) -> Option<Match> {
    ruleset
        .rules
        .par_iter()
        .flat_map(|rule| evaluate_rule(rule, log))
        .max_by(|a, b| a.rule.priority.cmp(&b.rule.priority))
}
