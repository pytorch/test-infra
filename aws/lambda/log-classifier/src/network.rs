//! Various utilities for interacting with the external world.
use anyhow::{Context, Result};
use aws_sdk_dynamodb as dynamodb;
use aws_sdk_s3 as s3;
use aws_sdk_s3::types::ByteStream;
use bytes::buf::Buf;
use bytes::Bytes;
use dynamodb::model::{AttributeAction, AttributeValueUpdate};
use flate2::read::GzDecoder;
use serde_dynamo::aws_sdk_dynamodb_0_18::to_attribute_value;
use std::io::Read;
use tracing::info;

use crate::rule_match::SerializedMatch;

static BUCKET_NAME: &str = "ossci-raw-job-status";

/// Creates an S3 client instance preconfigured with the right credentials/region.
pub async fn get_s3_client() -> s3::Client {
    let config = aws_config::from_env().region("us-east-1").load().await;
    s3::Client::new(&config)
}

pub async fn get_dynamo_client() -> dynamodb::Client {
    let config = aws_config::load_from_env().await;
    dynamodb::Client::new(&config)
}

/// Download a log for `job_id` from S3.
pub async fn download_log(client: &s3::Client, repo: &str, job_id: usize) -> Result<String> {
    let key = match repo {
       "pytorch/pytorch" => format!("log/{}", job_id),
       _ => format!("log/{}/{}", repo, job_id)
    };
    let resp = client
        .get_object()
        .bucket(BUCKET_NAME)
        .key(key)
        .send()
        .await?;

    let data = resp.body.collect().await?;
    let mut decoder = GzDecoder::new(data.reader());
    let mut raw_log = String::new();
    decoder
        .read_to_string(&mut raw_log)
        .context("failed to decompress log")?;
    Ok(raw_log)
}

/// Upload a classification for `job_id` to S3. `body` should be a serialized
/// instance of `SerializedMatch`.
pub async fn upload_classification_s3(
    client: &s3::Client,
    job_id: usize,
    body: String,
) -> Result<()> {
    client
        .put_object()
        .bucket(BUCKET_NAME)
        .key(format!("classification/{job_id}"))
        .content_type("application/json")
        .body(ByteStream::from(Bytes::from(body)))
        .send()
        .await?;
    info!("SUCCESS upload classification to s3 for job {}", job_id);
    Ok(())
}

/// Mutates the DynamoDB object for `job_id` to include the `SerializedMatch`.
pub async fn upload_classification_dynamo(
    client: &dynamodb::Client,
    repo: &str,
    job_id: usize,
    best_match: &SerializedMatch,
) -> Result<()> {
    let update = AttributeValueUpdate::builder()
        .action(AttributeAction::Put)
        .value(to_attribute_value(best_match)?)
        .build();
    client
        .update_item()
        .table_name("torchci-workflow-job")
        .key(
            "dynamoKey",
            to_attribute_value(format!("{}/{}", repo, job_id))?,
        )
        .attribute_updates("torchci_classification", update)
        .send()
        .await?;
    info!("SUCCESS upload classification to dynamo for job {}", job_id);
    Ok(())
}
