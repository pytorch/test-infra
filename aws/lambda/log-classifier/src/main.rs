use lambda_http::{run, service_fn, Body, Error, IntoResponse, Request, RequestExt, Response};

use anyhow::Result;
use std::time::Instant;
use tracing::info;

use log_classifier::engine::evaluate_ruleset;
use log_classifier::log::Log;
use log_classifier::network::{
    download_log, get_dynamo_client, get_s3_client, upload_classification_dynamo,
};
use log_classifier::rule::RuleSet;
use log_classifier::rule_match::SerializedMatch;

struct ShouldWriteDynamo(bool);

async fn handle(job_id: usize, repo: &str, should_write_dynamo: ShouldWriteDynamo) -> Result<String> {
    let client = get_s3_client().await;
    // Download the log from S3.
    let start = Instant::now();
    let raw_log = download_log(&client, repo, job_id).await?;
    info!("download: {:?}", start.elapsed());

    // Do some preprocessing.
    let start = Instant::now();
    let log = Log::new(raw_log);
    info!("preproc: {:?}", start.elapsed());

    // Run the matching
    let start = Instant::now();
    let ruleset = RuleSet::new_from_config();
    let maybe_match = evaluate_ruleset(&ruleset, &log);
    info!("evaluate: {:?}", start.elapsed());

    match maybe_match {
        Some(best_match) => {
            let match_json = SerializedMatch::new(&best_match, &log);
            let body = serde_json::to_string_pretty(&match_json)?;
            info!("match: {}", body);
            if should_write_dynamo.0 {
                let client = get_dynamo_client().await;
                upload_classification_dynamo(&client, repo, job_id, &match_json).await?;
            }
            Ok(body)
        }
        None => {
            info!("no match found for {}", job_id);
            Ok("No match found".into())
        }
    }
}

async fn function_handler(event: Request) -> Result<Response<Body>, Error> {
    // Extract some useful information from the request
    let query_string_parameters = event.query_string_parameters();
    Ok(match query_string_parameters.first("job_id") {
        Some(job_id) => {
            let job_id = job_id.parse::<usize>()?;
            let repo = query_string_parameters.first("repo").unwrap_or_else(|| "pytorch/pytorch");
            handle(job_id, repo, ShouldWriteDynamo(true))
                .await?
                .into_response()
                .await
        }

        _ => Response::builder()
            .status(400)
            .body("no job id provided".into())
            .expect("failed to render response"),
    })
}

#[tokio::main]
async fn main() -> Result<(), Error> {
    tracing_subscriber::fmt()
        .with_max_level(tracing::Level::INFO)
        // disabling time is handy because CloudWatch will add the ingestion time.
        .without_time()
        .init();

    run(service_fn(function_handler)).await
}

#[cfg(test)]
mod test {
    use super::*;
    use log_classifier::engine::evaluate_rule;
    use log_classifier::rule::Rule;
    use regex::Regex;

    #[test]
    fn basic_evaluate_rule() {
        let rule = Rule {
            name: "test".into(),
            pattern: r"^test".parse().unwrap(),
            priority: 100,
        };

        let log = Log::new("test foo".into());
        let match_ = evaluate_rule(&rule, &log);
        assert_eq!(match_.unwrap().line_number, 1);
    }

    #[test]
    fn escape_codes_are_stripped() {
        let mut ruleset = RuleSet::new();
        ruleset.add("foo", r"^test foo");
        let log = Log::new(
            "\
            2022-08-26T17:16:41.9362224Z \x1b[93;41mtest\x1b[0m foo\n\
            2022-08-26T17:16:41.9362224Z lol!lol\n\
            "
            .into(),
        );
        let match_ = evaluate_ruleset(&ruleset, &log).unwrap();
        assert_eq!(match_.line_number, 1);
        assert_eq!(match_.rule.name, "foo");
    }

    #[test]
    fn timestamp_is_stripped() {
        let mut ruleset = RuleSet::new();
        ruleset.add("foo", r"^test");
        let log = Log::new(
            "\
            2022-08-26T17:16:41.9362224Z test foo\n\
            2022-08-26T17:16:41.9362224Z lol!lol\n\
            "
            .into(),
        );
        let match_ = evaluate_ruleset(&ruleset, &log).unwrap();
        assert_eq!(match_.line_number, 1);
        assert_eq!(match_.rule.name, "foo");
    }

    #[test]
    fn evaluate_rulset_respects_priority() {
        let mut ruleset = RuleSet::new();
        ruleset.add("higher priority", r"^lol!");
        ruleset.add("lower priority", r"^test");
        let log = Log::new(
            "\
            test foo\n\
            lol!lol\n\
            "
            .into(),
        );
        let match_ = evaluate_ruleset(&ruleset, &log).unwrap();
        assert_eq!(match_.line_number, 2);
        assert_eq!(match_.rule.name, "higher priority");
    }

    #[test]
    fn ignore_skips_match() {
        let mut ruleset = RuleSet::new();
        ruleset.add("test", r"^test");
        let log = Log::new(
            "\
            =================== sccache compilation log ===================\n\
            testt\n\
            =========== If your build fails, please take a look at the log above for possible reasons ===========\n\
            "
                .into(),
        );
        let match_ = evaluate_ruleset(&ruleset, &log);
        assert!(match_.is_none());
    }

    #[test]
    fn match_before_ignore() {
        let mut ruleset = RuleSet::new();
        ruleset.add("test", r"^test");
        let log = Log::new(
            "\
            testt\n\
            =================== sccache compilation log ===================\n\
            =========== If your build fails, please take a look at the log above for possible reasons ===========\n\
            "
                .into(),
        );
        let match_ = evaluate_ruleset(&ruleset, &log).unwrap();
        assert_eq!(match_.line_number, 1);
    }

    #[test]
    fn match_after_ignore() {
        let mut ruleset = RuleSet::new();
        ruleset.add("test", r"^test");
        let log = Log::new(
            "\
            =================== sccache compilation log ===================\n\
            =========== If your build fails, please take a look at the log above for possible reasons ===========\n\
            testt\n\
            "
                .into(),
        );
        let match_ = evaluate_ruleset(&ruleset, &log).unwrap();
        assert_eq!(match_.line_number, 3);
    }

    #[test]
    fn later_match_wins() {
        let mut ruleset = RuleSet::new();
        ruleset.add("test", r"^test");
        let log = Log::new(
            "\
            testt\n\
            testt\n\
            "
            .into(),
        );
        let match_ = evaluate_ruleset(&ruleset, &log).unwrap();
        assert_eq!(match_.line_number, 2);
    }

    #[test]
    fn rules_compile_correctly() {
        // Try re-compiling the rules to make sure there are no invalid regexes.
        let ruleset = RuleSet::new_from_config();
        for rule in &ruleset.rules {
            Regex::new(rule.pattern.as_str()).unwrap();
        }
    }

    // Actually download some id.
    // #[tokio::test]
    // async fn test_real() {
    //    let foo = handle(12421522599, "pytorch/vision", ShouldWriteDynamo(false)).await;
    //    panic!("{:#?}", foo);
    // }
}
