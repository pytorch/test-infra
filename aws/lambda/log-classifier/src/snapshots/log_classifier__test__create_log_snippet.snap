---
source: src/main.rs
expression: result_string
---
  inflating: build/bin/wrapdim_test
  inflating: build/bin/xla_tensor_test
  inflating: build/bin/IListRef_test
  inflating: build/bin/List_test
  inflating: build/bin/KernelFunction_test
  inflating: build/bin/kernel_function_legacy_test
  inflating: build/bin/kernel_function_test
  inflating: build/bin/kernel_lambda_legacy_test
  inflating: build/bin/kernel_lambda_test
  inflating: build/bin/kernel_stackbased_test
  inflating: build/bin/make_boxed_from_unboxed_functor_test
  inflating: build/bin/CppSignature_test
  inflating: build/bin/op_allowlist_test
  inflating: build/bin/backend_fallback_test
  inflating: build/bin/op_registration_test
  inflating: build/bin/inline_container_test
  inflating: build/bin/cuda_apply_test
  inflating: build/bin/cuda_allocator_test
  inflating: build/bin/cuda_atomic_ops_test
  inflating: build/bin/cuda_caching_host_allocator_test
  inflating: build/bin/cuda_complex_math_test
  inflating: build/bin/cuda_complex_test
  inflating: build/bin/cuda_device_test
  inflating: build/bin/cuda_cub_test
  inflating: build/bin/cuda_dlconvertor_test
  inflating: build/bin/cuda_distributions_test
  inflating: build/bin/cuda_generator_test
  inflating: build/bin/cuda_half_test
  inflating: build/bin/cuda_integer_divider_test
  inflating: build/bin/cuda_optional_test
  inflating: build/bin/cuda_packedtensoraccessor_test
  inflating: build/bin/cuda_reportMemoryUsage_test
  inflating: build/bin/cuda_allocatorTraceTracker_test
  inflating: build/bin/cuda_stream_test
  inflating: build/bin/cuda_cudnn_test
  inflating: build/bin/cuda_vectorized_test
  inflating: build/bin/tutorial_tensorexpr
  inflating: build/bin/ProcessGroupGlooTest
  inflating: build/bin/ProcessGroupGlooAsyncTest
  inflating: build/bin/ProcessGroupNCCLTest
  inflating: build/bin/ProcessGroupNCCLErrorsTest
  inflating: build/bin/test_tensorexpr
  inflating: build/bin/test_jit
   creating: .additional_ci_files/
  inflating: .additional_ci_files/test-times.json
  inflating: .additional_ci_files/test-class-times.json
##[group]Run rm artifacts.zip
rm artifacts.zip
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  GIT_DEFAULT_BRANCH: main
  GPU_FLAG: --gpus all -e NVIDIA_DRIVER_CAPABILITIES=all
##[endgroup]
##[group]Run df -H
df -H
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  GIT_DEFAULT_BRANCH: main
  GPU_FLAG: --gpus all -e NVIDIA_DRIVER_CAPABILITIES=all
##[endgroup]
Filesystem      Size  Used Avail Use% Mounted on
devtmpfs         34G     0   34G   0% /dev
tmpfs            34G     0   34G   0% /dev/shm
tmpfs            34G  455k   34G   1% /run
tmpfs            34G     0   34G   0% /sys/fs/cgroup
/dev/nvme0n1p1  162G   43G  119G  27% /
Prepare all required actions
Getting action download info
##[group]Run ./.github/actions/download-td-artifacts
with:
env:
  GIT_DEFAULT_BRANCH: main
  GPU_FLAG: --gpus all -e NVIDIA_DRIVER_CAPABILITIES=all
##[endgroup]
##[group]Run seemethere/download-artifact-s3@v4
with:
  name: td_results
  s3-bucket: gha-artifacts
  region: us-east-1
env:
  GIT_DEFAULT_BRANCH: main
  GPU_FLAG: --gpus all -e NVIDIA_DRIVER_CAPABILITIES=all
##[endgroup]
(node:28537) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.
2024-07-08T16:56:59.7679407Z
Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the migration guide at https://a.co/7PzMCcy
(Use `node --trace-warnings ...` to show where the warning was created)
Found 0 objects with prefix pytorch/pytorch/9843060221/td_results/
Artifact download has finished successfully
##[group]Run mkdir -p .additional_ci_files
mkdir -p .additional_ci_files
mv td_results.json .additional_ci_files/td_results.json
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  GIT_DEFAULT_BRANCH: main
  GPU_FLAG: --gpus all -e NVIDIA_DRIVER_CAPABILITIES=all
##[endgroup]
mv: cannot stat 'td_results.json': No such file or directory
##[error]Process completed with exit code 1.
